diff --git a/config/riscv32/arch.cfg b/config/riscv32/arch.cfg
index 63d47af..72edb14 100644
--- a/config/riscv32/arch.cfg
+++ b/config/riscv32/arch.cfg
@@ -1,59 +1,23 @@
-# Architecture configuration for RISC-V
-#
-# Copyright (C) 2019 Embecosm Limited and the University of Bristol
-#
-# Contributor Graham Markall <graham.markall@embecosm.com>
-# Contributor Jeremy Bennett <jeremy.bennett@embecosm.com>
-#
-# This file is part of Embench and was formerly part of the Bristol/Embecosm
-# Embedded Benchmark Suite.
-#
-# SPDX-License-Identifier: GPL-3.0-or-later
+cc = 'COMPILER'
+ld = 'COMPILER'
+cflags = ['-c', '-v', '-march=DEF_MARCH', '-mabi=DEF_MABI']
+
+
+
+ldflags = ['-nostdlib', '-T', 'RTL_BASE/example_tb/core/custom/link.ld',
+          'RTL_BASE/example_tb/core/custom/crt0.S',
+          'RTL_BASE/example_tb/core/custom/syscalls.c',
+          'RTL_BASE/example_tb/core/custom/vectors.S',
+          '-I INCLUDE_DIR']
+user_libs = ['-lgcc', '-lc', '-lm', '-lgcc', '-lc', '-lm', '-lgcc', '-lc', '-lm']
+
 
-# This is a python setting of parameters for the architecture.  The following
-# parameters may be set (other keys are silently ignored).  Defaults are shown
-# in brackets
-# - cc ('cc')
-# - ld (same value as for cc)
-# - cflags ([])
-# - ldflags ([])
 # - cc_define_pattern ('-D{0}')
 # - cc_incdir_pattern ('-I{0}')
 # - cc_input_pattern ('{0}')
 # - cc_output_pattern ('-o {0}')
 # - ld_input_pattern ('{0}')
 # - ld_output_pattern ('-o {0}')
-# - user_libs ([])
 # - dummy_libs ([])
 # - cpu_mhz (1)
 # - warmup_heat (1)
-
-# The "flags" and "libs" parameters (cflags, ldflags, user_libs, dummy_libs)
-# should be lists of arguments to be passed to the compile or link line as
-# appropriate.  Patterns are Python format patterns used to create arguments.
-# Thus for GCC or Clang/LLVM defined constants can be passed using the prefix
-# '-D', and the pattern '-D{0}' would be appropriate (which happens to be the
-# default).
-
-# "user_libs" may be absolute file names or arguments to the linker. In the
-# latter case corresponding arguments in ldflags may be needed.  For example
-# with GCC or Clang/LLVM is "-l" flags are used in "user_libs", the "-L" flags
-# may be needed in "ldflags".
-
-# Dummy libs have their source in the "support" subdirectory. Thus if 'crt0'
-# is specified, there should be a source file 'dummy-crt0.c' in the support
-# directory.
-
-# There is no need to set an unused parameter, and this file may be empty to
-# set no flags.
-
-# Parameter values which are duplicated in architecture, board, chip or
-# command line are used in the following order of priority
-# - default value
-# - architecture specific value
-# - chip specific value
-# - board specific value
-# - command line value
-
-# For flags, this priority is applied to individual flags, not the complete
-# list of flags.
diff --git a/config/riscv32/chips/generic/chip.cfg b/config/riscv32/chips/generic/chip.cfg
index 6632c3e..74d35a7 100644
--- a/config/riscv32/chips/generic/chip.cfg
+++ b/config/riscv32/chips/generic/chip.cfg
@@ -67,4 +67,4 @@
 
 # - we garbage collect unused sections on linking
 
-cc = 'riscv32-unknown-elf-gcc'
+# cc = 'riscv32-unknown-elf-gcc'
diff --git a/src/aha-mont64/mont64.c b/src/aha-mont64/mont64.c
index faeccdb..d2241e8 100644
--- a/src/aha-mont64/mont64.c
+++ b/src/aha-mont64/mont64.c
@@ -18,7 +18,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 423
+#define LOCAL_SCALE_FACTOR 1
 
 /* Computes a*b mod m using Montgomery multiplication (MM). a, b, and m
 are unsigned numbers with a, b < m < 2**64, and m odd. The code does
diff --git a/src/crc32/crc_32.c b/src/crc32/crc_32.c
index a268113..4bf45c8 100644
--- a/src/crc32/crc_32.c
+++ b/src/crc32/crc_32.c
@@ -20,7 +20,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 170
+#define LOCAL_SCALE_FACTOR 1
 
 #include <stdlib.h>
 
diff --git a/src/cubic/basicmath_small.c b/src/cubic/basicmath_small.c
index a3d2839..5fa8be9 100644
--- a/src/cubic/basicmath_small.c
+++ b/src/cubic/basicmath_small.c
@@ -13,7 +13,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 10
+#define LOCAL_SCALE_FACTOR 1
 
 
 
diff --git a/src/edn/libedn.c b/src/edn/libedn.c
index 48a748d..3ce5871 100644
--- a/src/edn/libedn.c
+++ b/src/edn/libedn.c
@@ -37,7 +37,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 87
+#define LOCAL_SCALE_FACTOR 1
 
 
 #define N 100
diff --git a/src/huffbench/libhuffbench.c b/src/huffbench/libhuffbench.c
index f65e05a..749ba48 100644
--- a/src/huffbench/libhuffbench.c
+++ b/src/huffbench/libhuffbench.c
@@ -52,7 +52,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 11
+#define LOCAL_SCALE_FACTOR 1
 
 /* BEEBS heap is just an array */
 
diff --git a/src/matmult-int/matmult-int.c b/src/matmult-int/matmult-int.c
index 41b387e..ae0a463 100644
--- a/src/matmult-int/matmult-int.c
+++ b/src/matmult-int/matmult-int.c
@@ -38,7 +38,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 46
+#define LOCAL_SCALE_FACTOR 1
 #define UPPERLIMIT 20
 #define RANDOM_VALUE (RandomInteger ())
 #define ZERO 0
diff --git a/src/minver/libminver.c b/src/minver/libminver.c
index 8015229..39cb1ab 100644
--- a/src/minver/libminver.c
+++ b/src/minver/libminver.c
@@ -61,7 +61,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 555
+#define LOCAL_SCALE_FACTOR 1
 
 int minver (int row, int col, float eps);
 int mmul (int row_a, int col_a, int row_b, int col_b);
diff --git a/src/nettle-aes/nettle-aes.c b/src/nettle-aes/nettle-aes.c
index d85b9f6..9f7bb4d 100644
--- a/src/nettle-aes/nettle-aes.c
+++ b/src/nettle-aes/nettle-aes.c
@@ -20,7 +20,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 78
+#define LOCAL_SCALE_FACTOR 1
 
 // From nettle/macros.h
 
diff --git a/src/nettle-sha256/nettle-sha256.c b/src/nettle-sha256/nettle-sha256.c
index 28f53d6..1d81d38 100644
--- a/src/nettle-sha256/nettle-sha256.c
+++ b/src/nettle-sha256/nettle-sha256.c
@@ -20,7 +20,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 475
+#define LOCAL_SCALE_FACTOR 1
 
 // From nettle/nettle-types.h
 
diff --git a/src/nsichneu/libnsichneu.c b/src/nsichneu/libnsichneu.c
index 6952136..ec1513b 100644
--- a/src/nsichneu/libnsichneu.c
+++ b/src/nsichneu/libnsichneu.c
@@ -38,7 +38,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 1231
+#define LOCAL_SCALE_FACTOR 1
 
 #ifdef DO_TRACING		// ON PC
 
diff --git a/src/picojpeg/picojpeg_test.c b/src/picojpeg/picojpeg_test.c
index 134cde0..b7dcff5 100644
--- a/src/picojpeg/picojpeg_test.c
+++ b/src/picojpeg/picojpeg_test.c
@@ -18,7 +18,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 6
+#define LOCAL_SCALE_FACTOR 1
 
 const unsigned char jpeg_data[] = {
   0xff, 0xd8, 0xff, 0xe0, 0x00, 0x10, 0x4a, 0x46,
diff --git a/src/qrduino/qrtest.c b/src/qrduino/qrtest.c
index 2ce36ca..b53f953 100644
--- a/src/qrduino/qrtest.c
+++ b/src/qrduino/qrtest.c
@@ -20,7 +20,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 5
+#define LOCAL_SCALE_FACTOR 1
 
 /* BEEBS heap is just an array */
 
diff --git a/src/sglib-combined/combined.c b/src/sglib-combined/combined.c
index de41b86..d0c2bd8 100644
--- a/src/sglib-combined/combined.c
+++ b/src/sglib-combined/combined.c
@@ -17,7 +17,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 29
+#define LOCAL_SCALE_FACTOR 1
 
 /* BEEBS heap is just an array */
 
diff --git a/src/slre/libslre.c b/src/slre/libslre.c
index ee1353a..a80236f 100644
--- a/src/slre/libslre.c
+++ b/src/slre/libslre.c
@@ -18,7 +18,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 110
+#define LOCAL_SCALE_FACTOR 1
 
 #include <stdio.h>
 #include <ctype.h>
diff --git a/src/st/libst.c b/src/st/libst.c
index ce5b391..4817739 100644
--- a/src/st/libst.c
+++ b/src/st/libst.c
@@ -34,7 +34,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 13
+#define LOCAL_SCALE_FACTOR 1
 
 #define MAX 100
 
diff --git a/src/statemate/libstatemate.c b/src/statemate/libstatemate.c
index 2bf8e36..e869e62 100644
--- a/src/statemate/libstatemate.c
+++ b/src/statemate/libstatemate.c
@@ -46,7 +46,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 1964
+#define LOCAL_SCALE_FACTOR 1
 
 /*
 ** actually, we don't really need floating point here
diff --git a/src/ud/libud.c b/src/ud/libud.c
index a7a85bd..d14ebfc 100644
--- a/src/ud/libud.c
+++ b/src/ud/libud.c
@@ -90,7 +90,7 @@
 
 /* This scale factor will be changed to equalise the runtime of the
    benchmarks. */
-#define LOCAL_SCALE_FACTOR 1478
+#define LOCAL_SCALE_FACTOR 1
 
 
 long int a[20][20], b[20], x[20];
